src/core/filesystem.cc

```diff
@@ -1175,6 +1180,40 @@ S3FileSystem::ParsePath(
   return Status::Success;
 }
 
+Status
+S3FileSystem::CleanPath(const std::string& path, std::string* clean_path)
+{
+  // Remove extra slashes
+  *clean_path = "s3://";
+  int true_length = path.size();
+  for (size_t i = path.size() - 1; i++) {
+    if (path[i] == '/') {
+      true_length -= 1;
```

Comment:
This is going do reduce length for internal slashes, isn't it? See comment below for an easier solution.

Reply:
Yes